service/controller/azuremachine/handler/azuremachinemetadata/create.go

```diff
@@ -0,0 +1,41 @@
+package azuremachinemetadata
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/giantswarm/azure-operator/v5/pkg/helpers"
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, cr interface{}) error {
+	var err error
+	azureMachine, err := key.ToAzureMachine(cr)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// We need this for existing clusters, and we do it both in Cluster
+	// controller (clusterupgrade) and in AzureMachine controller, because we
+	// do not know which controller/handler will be executed first, and we need
+	// annotation set correctly in both places.
+	err = helpers.InitAzureMachineAnnotations(ctx, r.ctrlClient, r.logger, &azureMachine)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ctrlClient.Update(ctx, &azureMachine)
```

Comment:
So basically we update the cr at every reconciliation loop, right?
Even if there are no changes being made.
If the update call makes a change in the timestamps we're going to end up with an infinite loop maybe?

Reply:
If there were no actual changes in the CR, nothing should happen, i.e. update call would have no effect. We have code like this in many places in azure-operator.

Reply:
good